public/js/components/card-input.jsx

```diff
@@ -98,14 +98,23 @@ export default class CardInput extends Component {
                       errorModifier={this.props.errorModifier} /> : null }
         { showCardIcon ?
           <PayMethodIcon payMethodType={this.props.cardType} /> : null }
-        <MaskedInput
-          {...this.props.attrs}
-          id={this.props.id}
-          className={this.props.id + '-input'}
-          onChange={this.props.onChangeHandler}
-          pattern={pattern}
-          placeholder={placeholder}
-        />
```

Comment:
I guess this could also be:
````
var extraProps = {};
if (pattern) {
  extraProps.pattern = pattern;
}
<MaskedInput
            {...this.props.attrs}
            {...extraProps}
            id={this.props.id}
            className={this.props.id + '-input'}
            onChange={this.props.onChangeHandler}
            placeholder={placeholder}
          />
````

that way you wouldn't have to repeat so many fields.